internals/feature_links.py

```diff
@@ -211,3 +212,57 @@ def batch_index_feature_entries(fes: list[FeatureEntry], skip_existing: bool) ->
     logging.info(f'Feature {fe.key.integer_id()} indexed {len(feature_links)} urls')
 
   return link_count
+
+def get_feature_links_summary():
+  """
+  The function `get_feature_links_summary` retrieves feature links from a database, groups them by
+  type and uncovered domains, and returns a summary of the counts and types of links.
+  """
+  feature_links = FeatureLinks.query().fetch(
+      projection=[
+          FeatureLinks.feature_ids,
+          FeatureLinks.url,
+          FeatureLinks.type,
+          FeatureLinks.is_error,
+          FeatureLinks.http_error_code,
+      ]
+  )
+  links = [item.to_dict() for item in feature_links]
+
+  def group_by(list, get_key_fn):
+      dict = {}
+      for item in list:
+          try:
+              key = get_key_fn(item)
+              collection = dict.get(key) or []
+              collection.append(item)
+              dict[key] = collection
+          except:
+              continue
+      return dict
+  
+  def count_and_sort(dict, key_name = 'key'):
+      list = []
+      for k, v in dict.items():
+          list.append({
+              key_name: k,
+              "count": len(v),
+          })
+      list.sort(key=lambda x: x["count"], reverse=True)
+      return list
+  
+  def get_domain_with_scheme(url):
+    scheme, host = urlparse(url).scheme, urlparse(url).netloc
+    return f"{scheme}://{host}"
+  
+  uncovered_links = [link for link in links if link['type'] == 'web']
+  link_types = group_by(links, lambda item: item['type'])
```

Comment:
If you use `collections.Counter`, then this could be something like:
```
type_counter = Counter(item['type'] for item in links)
link_types = [
    {'key': k, 'count': c} 
    for (k, c) in type_counter.most_common().items()]

domain_counter = Counter(get_domain_with_scheme(item['url']) for item in uncovered_links)
uncommon_link_domains = [
    {'key': k, 'count': c} 
    for (k, c) in domain_couner.most_common().items()]

return { 
    ... other fields ...
   "link_types": link_types,
   "uncommon_link_domains": uncommon_link_domains,
}
```

And you would not need the 2 helper functions that you wrote.